### ISSUE TITLE
Unable to connect to Server with certain clients.

### ISSUE THREAD

**Issue by kevinpostal** [DanielOaks/oragono#74]
I am unable to connect to Oragono server using limechat. http://limechat.net/

The clients seems to connect but freezes on Looking up your username.

```
Connected
*** Looking up your username
```

**Comment by kevinpostal** [comment]
It seems like my client will connect after about a Minute of Waiting.

**Comment by DanielOaks** [comment]
Hmm that's sketchy, I'll setup a test server and try connecting to it. Just to make sure by chance, did you end up updating go-ident with a command like this?

```
go get -u github.com/DanielOaks/go-ident
```

Regardless, I'll give it a shot and see what I find out, thanks for letting me know!

**Comment by DanielOaks** [comment]
Actually yeah, I can repro this issue locally. I'll fix this and let you know once it's resolved, thanks.

**Comment by kevinpostal** [comment]
Thanks so much @DanielOaks ! I really hope to use this IRCD In production!

**Comment by DanielOaks** [comment]
Alright so there were a good few issues around this, I'll go over them one-by-one.

1. go-ident wasn't timing out dialing, it now does thanks to https://github.com/DanielOaks/go-ident/commit/57ead167851eab48dddf0a57f0ba8d838346f3ba
2. ircmsg was failing to push out the right messages because of a bug I introduced, fixed the bug and added a test for it in https://github.com/DanielOaks/girc-go/commit/cfa836b1b08b91b2961783777278ce727dd0caef
3. cmode changes weren't being conveyed to all members in the channel, fixed by 9314edd405ca80afb92563f7d4f1db7762408e35
4. quit messages weren't being sent correctly when other clients in channels disconnected, fixed by 2e2e91689a5aaca133e5f989a830ca7f421b36db

So this and #75 should now be fixed.

To make sure you're running all of the newest code, could I please ask you to run:
```
go get -u github.com/DanielOaks/go-ident
go get -u github.com/DanielOaks/girc-go
```

And update to the newest `master` commit in this repo? After that, you may have to compile and run this:
```
oragono upgradedb
```

Then you should be able to start the IRCd correctly.

Thanks again for raising these issues with me, it's much appreciated! Lots of very good, necessary, useful fixes being pushed because of the issues discovered here. Let me know whether these issues are resolved through these updates, or whether there are additional troubles you run into.

**Comment by DanielOaks** [comment]
So I think this should be all fixed thanks to those changes. If you could test them out and give me a yell, it'd be much appreciated :)
